src/mlpack/tests/ann_visitor_test.cpp

```diff
@@ -53,6 +53,41 @@ TEST_CASE("BiasSetVisitorTest", "[ANNVisitorTest]")
   boost::apply_visitor(DeleteVisitor(), linear);
 }
 
+/**
+ * Test that WeightSizeVisitor works properly for Convolution layer.
+ */
+TEST_CASE("WeightSizeVisitorTestForConvLayer", "[ANNVisitorTest]")
+{
+  size_t randomInSize = arma::randi(arma::distr_param(1, 100));
+  size_t randomOutSize = arma::randi(arma::distr_param(1, 100));
+  size_t randomKernelWidth = arma::randi(arma::distr_param(1, 100));
+  size_t randomKernelHeight = arma::randi(arma::distr_param(1, 100));
+
+  LayerTypes<> convLayer = new Convolution<>(randomInSize, randomOutSize,
+      randomKernelWidth, randomKernelHeight);
+
+  size_t weightSize = boost::apply_visitor(WeightSizeVisitor(),
+                                           convLayer);
+
+  REQUIRE(weightSize == (randomOutSize * randomInSize * randomKernelWidth *
+      randomKernelHeight) + randomOutSize);
```

Comment:
Might be better to test against `convLayer.Parameters().n_elem`?

Reply:
Yes that would be better. Made the changes to reflect the same. Thanks for the review.